### DISCUSSION TITLE
How to validate data and show error before running workflow?

### DISCUSSION THREAD

**Discussion by nihilne** [YetiForceCompany/YetiForce#68] [How to] [answered]
Hello there,

I have a custom workflow, and I want to check if the data that the user has entered in a record is correct. I want to do that check before the custom workflow is run.

Is there a way to show an error message to the user if some of the data is not correct? I don't know if it's possible inside the custom workflow, or with some sort of event handler except for editViewPreSave. Ideally I want it to be inside the custom workflow, and I can send an error if something goes wrong.

Thank you very much!

**Comment by rskrzypczak** [comment, marked as answer]
You can't do this through workflow.
Workflow starts after the record is saved, so it's too late.
As you mentioned, the only solution is to add a handler and only `editViewPreSave` is able to display feedback to the user.

**Reply by nihilne** [reply]
Ah I see, thank you very much!
